app/services/bing/parse_result_service.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Bing
+  class ParseResultService
+    TOP_ADS_SECTION_CLASS = 'b_ad b_adTop'
+    ADS_CLASS = 'sb_add sb_adTA'
+    SIDE_BANNER_ADS_CONTAINER_CLASS = 'sb_add sb_adTA sb_adBrandSidebar ad_sc'
+    SIDE_BANNER_ADS_IMAGE_CLASS = 'sb_adImageExtensionFull'
+    SEARCH_RESULT_CONTAINER_ID = 'b_results'
+    SEARCH_RESULT_CLASS = 'b_algo'
+    ADS_LINK_ROLE = 'link'
+
+    def initialize(html)
+      @html = html
+    end
+
+    def call
+      @doc = Nokogiri::HTML(html)
+
+      {
+        ads_top_count: ads_top_count,
+        ads_page_count: ads_page_count,
+        ads_top_urls: ads_top_urls,
+        result_count: result_count,
+        result_urls: result_urls,
+        total_link_count: total_link_count,
+        html: html
+      }
+    end
+
+    private
+
+    attr_reader :html, :doc
```

Comment:
It's just optional, because when reading these attributes. i feel a bit confuse about what the different between `:html` and `:doc`

```suggestion
    attr_reader :raw_html, :parsed_html
```